lib/lf/geometry/segment_o2.cc

```diff
@@ -0,0 +1,171 @@
+/**
+ * @file
+ * @brief Implementation of second-order segments
+ * @author Anian Ruoss
+ * @date   2018-11-18 19:02:17
+ * @copyright MIT License
+ */
+
+#include "segment_o2.h"
+#include "point.h"
+
+namespace lf::geometry {
+
+Eigen::MatrixXd SegmentO2::Global(const Eigen::MatrixXd& local) const {
+  Eigen::MatrixXd global(DimGlobal(), local.cols());
+
+  const Eigen::VectorXd vtx0 = coords_.col(0);
+  const Eigen::VectorXd vtx1 = coords_.col(1);
+  const Eigen::VectorXd midp = coords_.col(2);
+
+  // polynomial of degree 2: alpha * x^2 + beta * x + gamma
+  const Eigen::VectorXd alpha = 2. * (vtx1 + vtx0) - 4. * midp;
+  const Eigen::VectorXd beta = 4. * midp - 3. * vtx0 - vtx1;
+  const Eigen::VectorXd& gamma = vtx0;
+
```

Comment:
- No loop; use Eigen::Array operations for compact and efficient code.
- Use Horner scheme for evaluation of polynomial.